File: app/src/main/java/com/katkov/lolachievements/prefser/EntryInfoHolder.java
package com.katkov.lolachievements.prefser;

import com.github.pwittchen.prefser.library.rx2.Prefser;
import com.katkov.lolachievements.domain.model.EntryInfoModel;

import javax.inject.Inject;

public class EntryInfoHolder {

    private static final String ENTRY_INFO_KEY = "entry_info_key";

    private final Prefser prefser;

    @Inject
    public EntryInfoHolder(Prefser prefser) {
        this.prefser = prefser;
    }

    public void putEntryInfo(EntryInfoModel entryInfoModel) {
        prefser.put(ENTRY_INFO_KEY, entryInfoModel);
    }

    public EntryInfoModel getEntryInfoModel() {
        return prefser.get(ENTRY_INFO_KEY, EntryInfoModel.class, null);
    }

    public void removeEntryInfo() {
        prefser.remove(ENTRY_INFO_KEY);
    }
}

File: app/src/main/java/com/katkov/lolachievements/data/local/prefser/EntryInfoHolder.kt
package com.katkov.lolachievements.data.local.prefser

import com.github.pwittchen.prefser.library.rx2.Prefser
import com.katkov.lolachievements.domain.model.EntryInfoModel

import javax.inject.Inject

class EntryInfoHolder
@Inject
constructor(private val prefser: Prefser) {

    fun getEntryInfo(): EntryInfoModel? {
        return prefser.get(ENTRY_INFO_KEY, EntryInfoModel::class.java, null)
    }

    fun putEntryInfo(entryInfoModel: EntryInfoModel) {
        prefser.put(ENTRY_INFO_KEY, entryInfoModel)
    }

    fun removeEntryInfo() {
        prefser.remove(ENTRY_INFO_KEY)
    }

    companion object {

        private const val ENTRY_INFO_KEY = "entry_info_key"
    }
}
